skfda/representation/basis/_fdatabasis.py

```diff
@@ -356,6 +357,43 @@ def derivative(self: T, *, order: int = 1) -> T:  # noqa: D102
 
         return self.copy(basis=basis, coefficients=coefficients)
 
+    def integrate(
+        self: T,
+        *,
+        interval: Optional[DomainRange] = None,
+    ) -> NDArrayFloat:
+        """Examples.
+
+
+        We first create the data basis.
+            >>> from skfda.representation.basis import FDataBasis, Monomial
+            >>> basis = Monomial(n_basis=4)
+            >>> coefficients = [1, 1, 3, .5]
+            >>> fdata = FDataBasis(basis, coefficients)
+
+        Then we can integrate on the whole domain.
+            >>> fdata.integrate()
+            array([[ 2.625]])
+
+        Or we can do it on a given domain.
+            >>> fdata.integrate(interval = ((0.5, 1),))
+            array([[ 1.8671875]])
+
+        """
+        if interval is None:
+            interval = self.basis.domain_range
+
+        integrated = nquad_vec(
+            self,
+            interval,
+        )
+        integrated_values = np.empty((1, 1))
+
+        for i in integrated:
+            integrated_values = np.concatenate([integrated_values, i])
+
+        return integrated_values[1:]
```

Comment:
I don't get what are you trying to do here. Care to explain?

Reply:
Actually, I don't remember. But you are right, it does not make any sense